File: pset6/readability/readability.py
from cs50 import get_string

# Get user input
s = get_string("Text: ")

# Initialize letter word sentences
letters = 0
words = 1
sentences = 0

# Loop over the string
for i in range(len(s)):
    if s[i].isalpha():
        letters += 1
    if s[i].isspace():
        words += 1
    if s[i] == '.' or s[i] == '!' or s[i] == '?':
        sentences += 1

# Calculate grade level
L = letters / (words / 100)
S = sentences / (words / 100)
I = round((0.0588 * L) - (0.296 * S) - 15.8)

# Print grade level
if I < 1:
    print("Before Grade 1")

elif I > 16:
    print("Grade 16+")
    
elif I >= 1 or I <= 16:
    print(f"Grade {I}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string s = get_string("Text: ");                          // Get user Input                                              
   
    int letters = 0;                                           // Count number of letters
    int words = 1;                                               // Count number of words
    int sentences = 0;                                          // Count sentences
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (isalpha(s[i]))
        {
            letters++;
        }
        if (isspace(s[i]))
        {
            words++;
        }
        if (s[i] == '.' || s[i] == '!'  || s[i] == '?')
        {
            sentences++;
        }   
     
    }
   
   
   
    float L = letters / (words / 100.0) ;                       // Applying formula and defining variables
    float S = sentences / (words / 100.0);
    int I = round((0.0588 *  L) - (0.296 * S) - 15.8);
   
    if (I < 1)
    {
        printf("Before Grade 1\n");
    }                                                  // Loop for grade level
    if (I > 16)
    {
        printf("Grade 16+\n");
    }
    else if (I >= 1 && I <= 16)
    {
        printf("Grade %i\n", I);                        //Print grades
    }
   
}
 
 

 
